fission/src/mirabuf/EjectableSceneObject.ts

```diff
@@ -80,9 +80,9 @@ class EjectableSceneObject extends SceneObject {
             World.physicsSystem.disablePhysicsForBody(this._gamePieceBodyId)
 
             // Remove from any scoring zones
-            const zones = [...World.sceneRenderer.sceneObjects.entries()]
-                .filter(x => x[1] instanceof ScoringZoneSceneObject)
-                .map(x => x[1]) as ScoringZoneSceneObject[]
+            const zones = [...World.sceneRenderer.sceneObjects.values()].filter(
+                x => x instanceof ScoringZoneSceneObject
+            )
```

Comment:
I wouldn't be against giving a method like `filterSceneObjects(predicate)` to `sceneRenderer`.